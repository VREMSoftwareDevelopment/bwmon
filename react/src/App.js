/*
 *      Copyright (C) 2010 - 2020 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { HashRouter } from 'react-router-dom';
import { DateTime } from 'luxon';
import theme from './theme';
import { ThemeProvider } from '@material-ui/core/styles';
import { CssBaseline } from '@material-ui/core';
import Header from './components/main/Header';
import Footer from './components/main/Footer';
import Navigation from './components/navigation/Navigation';
import BWMonRoutes from './components/navigation/BWMonRoutes';
import ErrorBoundary from './components/error/ErrorBoundary';
import menu from './menu/Menu';

import { createBrowserHistory } from 'history';

export const history = createBrowserHistory({ basename: process.env.PUBLIC_URL });

const App = ({ name, version, currentTime }) => (
    <HashRouter>
        <ThemeProvider theme={theme}>
            <CssBaseline />
            <Header name={name} version={version} />
            <Navigation menu={menu} />
            <ErrorBoundary>
                <BWMonRoutes menu={menu} />
                <Footer currentTime={currentTime} />
            </ErrorBoundary>
        </ThemeProvider>
    </HashRouter>
);

App.defaultProps = {
    name: process.env.REACT_APP_NAME.toUpperCase(),
    description: process.env.REACT_APP_DESCRIPTION,
    version: process.env.REACT_APP_VERSION,
    currentTime: DateTime.local().toLocaleString(DateTime.DATETIME_FULL_WITH_SECONDS),
};

export default App;
