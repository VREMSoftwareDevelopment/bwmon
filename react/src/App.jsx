/*
 *      Copyright (C) 2010 - 2025 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import PropTypes from 'prop-types';
import { HashRouter } from 'react-router';
import { DateTime } from 'luxon';
import theme from './theme';
import { ThemeProvider, CssBaseline } from '@mui/material';
import { ErrorBoundary } from 'react-error-boundary';
import ErrorDisplay from './components/main/ErrorDisplay';
import Header from './components/main/Header';
import Footer from './components/main/Footer';
import Navigation from './components/navigation/Navigation';
import BWMonRoutes from './components/navigation/BWMonRoutes';
import menu from './menu/Menu';
import { createBrowserHistory } from 'history';

export const history = createBrowserHistory({ basename: import.meta.env.BASE_URL });

const appName = import.meta.env.VITE_DESCRIPTION || 'Bandwidth Monitor';
const appVersion = import.meta.env.VITE_VERSION;
const appTime = DateTime.local().toLocaleString(DateTime.DATETIME_FULL_WITH_SECONDS);

const App = ({ name = appName, version = appVersion, currentTime = appTime }) => {
    return (
        <HashRouter>
            <ThemeProvider theme={theme}>
                <CssBaseline />
                <Header name={name} version={version} />
                <Navigation menu={menu} />
                <ErrorBoundary onError={ErrorDisplay}>
                    <BWMonRoutes menu={menu} />
                    <Footer currentTime={currentTime} />
                </ErrorBoundary>
            </ThemeProvider>
        </HashRouter>
    );
};

App.propTypes = { name: PropTypes.string, version: PropTypes.string, currentTime: PropTypes.string };

export default App;
