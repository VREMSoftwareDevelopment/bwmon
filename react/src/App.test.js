/*
 *      Copyright (C) 2010 - 2020 VREM Software Development <VREMSoftwareDevelopment@gmail.com>
 *
 *      Licensed under the Apache License, Version 2.0 (the "License");
 *      you may not use this file except in compliance with the License.
 *      You may obtain a copy of the License at
 *
 *           http: //www.apache.org/licenses/LICENSE-2.0
 *
 *      Unless required by applicable law or agreed to in writing, software
 *      distributed under the License is distributed on an "AS IS" BASIS,
 *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *      See the License for the specific language governing permissions and
 *      limitations under the License.
 *
 * Bandwidth Monitor
 */

import React from 'react';
import { create } from 'react-test-renderer';
// import { act, create } from 'react-test-renderer';
// import wait from './__test__/utils/Wait';
import App from './App';

jest.mock('./services/Usage');

describe('App', () => {
    const name = 'BWMon';
    const version = '3.1.1';
    const currentTime = 'October 20, 2020, 11:25:35 AM EDT';

    test('renders correctly', () => {
        const tree = create(<App name={name} version={version} currentTime={currentTime} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    // current issue in Material UI Table Pagination label id
    // test('renders correctly after data load', async () => {
    //     let tree;
    //     act(() => {
    //         tree = create(<App name={name} version={version} currentTime={currentTime} />);
    //     });
    //     await wait(1);
    //     expect(tree.toJSON()).toMatchSnapshot();
    // });
});
