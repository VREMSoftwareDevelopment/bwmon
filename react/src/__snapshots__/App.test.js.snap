// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root makeStyles-title-1 MuiTypography-h6"
        id="app-title"
      >
        BWMon
         
        
      </h6>
      <p
        className="MuiTypography-root MuiTypography-body1"
        id="app-version"
      >
        3.1.1
      </p>
    </div>
  </header>,
  <div
    className="MuiBottomNavigation-root"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiBottomNavigationAction-root Mui-selected"
      href="#/UsageByUser"
      id="usage-by-user"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label Mui-selected"
        >
          By User
        </span>
      </span>
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiBottomNavigationAction-root"
      href="#/UsageByUserGraph"
      id="usage-by-user-graph"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3.5 18.49l6-6.01 4 4L22 6.92l-1.41-1.41-7.09 7.97-4-4L2 16.99z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label"
        >
          By User Graph
        </span>
      </span>
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiBottomNavigationAction-root"
      href="#/UsageByMonth"
      id="usage-by-month"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label"
        >
          By Month
        </span>
      </span>
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiBottomNavigationAction-root"
      href="#/UsageByMonthGraph"
      id="usage-by-month-graph"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M23 8c0 1.1-.9 2-2 2-.18 0-.35-.02-.51-.07l-3.56 3.55c.05.16.07.34.07.52 0 1.1-.9 2-2 2s-2-.9-2-2c0-.18.02-.36.07-.52l-2.55-2.55c-.16.05-.34.07-.52.07s-.36-.02-.52-.07l-4.55 4.56c.05.16.07.33.07.51 0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2c.18 0 .35.02.51.07l4.56-4.55C8.02 9.36 8 9.18 8 9c0-1.1.9-2 2-2s2 .9 2 2c0 .18-.02.36-.07.52l2.55 2.55c.16-.05.34-.07.52-.07s.36.02.52.07l3.55-3.56C19.02 8.35 19 8.18 19 8c0-1.1.9-2 2-2s2 .9 2 2z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label"
        >
          By Month Graph
        </span>
      </span>
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiBottomNavigationAction-root"
      href="#/UsageByYear"
      id="usage-by-year"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 17h18v2H3zm0-7h18v5H3zm0-4h18v2H3z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label"
        >
          By Year
        </span>
      </span>
    </a>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiBottomNavigationAction-root"
      href="#/UsageByYearGraph"
      id="usage-by-year-graph"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiBottomNavigationAction-wrapper"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M23 8c0 1.1-.9 2-2 2-.18 0-.35-.02-.51-.07l-3.56 3.55c.05.16.07.34.07.52 0 1.1-.9 2-2 2s-2-.9-2-2c0-.18.02-.36.07-.52l-2.55-2.55c-.16.05-.34.07-.52.07s-.36-.02-.52-.07l-4.55 4.56c.05.16.07.33.07.51 0 1.1-.9 2-2 2s-2-.9-2-2 .9-2 2-2c.18 0 .35.02.51.07l4.56-4.55C8.02 9.36 8 9.18 8 9c0-1.1.9-2 2-2s2 .9 2 2c0 .18-.02.36-.07.52l2.55 2.55c.16-.05.34-.07.52-.07s.36.02.52.07l3.55-3.56C19.02 8.35 19 8.18 19 8c0-1.1.9-2 2-2s2 .9 2 2z"
          />
        </svg>
        <span
          className="MuiBottomNavigationAction-label"
        >
          By Year Graph
        </span>
      </span>
    </a>
  </div>,
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="makeStyles-root-2"
    >
      <div
        className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
        role="progressbar"
      >
        <div
          className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
          style={Object {}}
        />
        <div
          className="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
          style={Object {}}
        />
      </div>
      <div
        className="makeStyles-root-3"
      >
        <div
          className="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
          role="alert"
        >
          <div
            className="MuiAlert-icon"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
              />
            </svg>
          </div>
          <div
            className="MuiAlert-message"
          >
            <div
              className="MuiTypography-root MuiAlertTitle-root MuiTypography-body1 MuiTypography-gutterBottom"
            >
              Loading...
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiTableContainer-root"
    />
  </div>,
  <div
    className="makeStyles-root-4"
  >
    <h6
      className="MuiTypography-root MuiTypography-subtitle2"
    >
      All usage information is in gigabytes
    </h6>
    <h6
      className="MuiTypography-root MuiTypography-subtitle2"
    >
      This page was generated on 
      October 20, 2020, 11:25:35 AM EDT
    </h6>
  </div>,
]
`;
